File: CodePlus/class2_/sohan/9012.py
N = int(input())
strings = [list(input()) for i in range(N)]
for string in strings:
    end = 0
    stack = []
    for char in string:
        if char == '(':
            stack.append(char)
        else:
            if len(stack) == 0:
                print("NO")
                end = -1
                break
            stack.pop()
    if (end != -1):
        print("YES" if len(stack) == 0 else "NO")

File: boj/mtak/_9012.java
import java.util.Scanner;
import java.util.Stack;

public class _9012 {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int t = scanner.nextInt();
        scanner.nextLine();
        for (int i = 0; i < t; i++) {
            Stack<String> stack = new Stack<>();
            char[] s = scanner.nextLine().toCharArray();
            boolean flag = false;
            for (char c : s) {
                if (c == '(') {
                    stack.push(String.valueOf(c));
                } else {
                    if (stack.isEmpty()) {
                        flag = true;
                        break;
                    }
                    stack.pop();
                }
            }
            if (!stack.isEmpty() || flag) {
                System.out.println("NO");
                continue;
            }
            System.out.println("YES");
        }
    }
}
